torch/distributed/elastic/rendezvous/c10d_rendezvous_backend.py

```diff
@@ -0,0 +1,195 @@
+        """Gets the `Store` instance used to communicate with the C10d store."""
+        return self._store
+
+    @property
+    def key(self) -> str:
+        """Gets the key under which the rendezvous state is stored."""
+        return self._key
+
+    def get_state(self) -> Optional[Tuple[bytes, Token]]:
+        """See base class."""
+        base64_state: bytes = self._call_store("get", self._key)
+
+        return self._decode_state(base64_state)
+
+    def set_state(
+        self, state: bytes, token: Optional[Token] = None
+    ) -> Optional[Tuple[bytes, Token]]:
+        """See base class."""
+        base64_state_str: str = codecs.encode(state, "base64").decode()
+
+        if token:
+            # Shortcut if we know for sure that the token is not valid.
+            if not isinstance(token, bytes):
+                return self.get_state()
+
+            token = token.decode()
+        else:
+            token = self._NULL_SENTINEL
+
+        base64_state: bytes = self._call_store("compare_set", self._key, token, base64_state_str)
+
+        return self._decode_state(base64_state)
+
+    def _call_store(self, store_op: str, *args, **kwargs) -> Any:
+        try:
+            return getattr(self._store, store_op)(*args, **kwargs)
+        except (ValueError, RuntimeError) as exc:
+            raise RendezvousConnectionError(
+                "The connection to the C10d store has failed. See inner exception for details."
+            ) from exc
+
+    def _decode_state(self, base64_state: bytes) -> Optional[Tuple[bytes, Token]]:
+        if base64_state == self._NULL_SENTINEL.encode():
+            return None
+        try:
+            state = codecs.decode(base64_state, "base64")
+        except binascii.Error as exc:
+            raise RendezvousStateError(
+                "The state object is corrupt. See inner exception for details."
+            ) from exc
+
+        return state, base64_state
+
+
+def _create_tcp_store(params: RendezvousParameters) -> TCPStore:
+    host, port = _parse_rendezvous_endpoint(params.endpoint, default_port=29500)
+
+    cfg_is_host = params.get_as_bool("is_host")
+    # If the user has explicitly specified whether our process should host the
+    # the store, respect it.
+    if cfg_is_host is not None:
+        is_host = cfg_is_host
+    # Otherwise try to determine whether we are the host based on our hostname
+    # and IP address.
+    else:
+        is_host = _matches_machine_hostname(host)
+
+    # The timeout
+    read_timeout = cast(int, params.get_as_int("read_timeout", 60))
+    if read_timeout <= 0:
+        raise ValueError("The read timeout must be a positive integer.")
+
+    # In specific cases we attempt to instantiate the store twice. For details
+    # see the explanation in the except clause below.
+    for _ in range(2):
+        try:
+            store = TCPStore(  # type: ignore[call-arg]
+                host, port, is_master=is_host, timeout=timedelta(seconds=read_timeout)
```

Comment:
Don't we need the ability to pass the world_size here as well?

Reply:
world_size is determined after a round of rendezvous is complete. This `TCPStore` instance is only used by the TorchElastic agents for synchronization between them. The "actual" store used by workers is later handled using a different mechanism inside an agent.